### PR TITLE
CI testing of newly integrated tool schema (XSD).

### PR DESCRIPTION
**Pull request by jmchilton**
- Add CI script that lints the XSD definition and then validates all test tools against it.
- Add tox environment to run the aforementioned script.
- Add Travis check through tox to run the script.
- Drop the multi_page.xml test tool - pages have been removed from the framework and are not part of the schema (xref https://github.com/galaxyproject/galaxy/issues/1776).

This will hopefully ensure that all new tool features get added to the schema (as long as there is a test tool for the feature). xref https://github.com/galaxyproject/galaxy/pull/2923 https://github.com/galaxyproject/galaxy/pull/2903

Largely based on work done in Galaxy-XSD by JeanFred (e.g. https://github.com/galaxyproject/Galaxy-XSD/commit/386ba087881877093d4be1f6d8547166f6163f3c)

### CODE REVIEW

**Comment by nsoranzo** [comment]
What about moving the main validation code from `.ci/validate_test_tools.sh` to another Bash script in `scripts/`, leaving only something like:

```
#!/bin/bash

cd `dirname $0`/..

xsd_path="lib/galaxy/tools/xsd/galaxy.xsd"
# Lint the XSD
xmllint --noout "$xsd_path"

test_tools_path='test/functional/tools'
tool_files_list=$(ls "$test_tools_path"/*xml | grep -v '_conf.xml$')
./scripts/tool_xsd_validator.py $tool_files_list
```

**Comment by jmchilton** [comment]
I don't want to compete with Planemo :cry: - I can make this change though.

**Comment by jmchilton** [comment]
I believe 05ae995 is as you requested @nsoranzo.

File: .travis.yml [hunk 1]
```diff
@@ -27,6 +28,7 @@ before_install:
 
 install:
   - pip install tox
+  - if [ "$TOX_ENV" == "validate-test-tools" ]]; then apt-get install libxml2-utils; fi
```
Comment (nsoranzo) [inline .travis.yml]: There's an extra `]` here, as reported by https://travis-ci.org/galaxyproject/galaxy/jobs/160534044#L162

File: tox.ini [hunk 2]
```diff
@@ -66,3 +66,7 @@ whitelist_externals = bash
 [testenv:first_startup]
 commands = bash .ci/first_startup.sh
 whitelist_externals = bash
+
+[testenv:validate_test_tools]
```
Comment (nsoranzo) [inline tox.ini]: This should be:

```
[testenv:validate-test-tools]
```

**Review by nsoranzo — CHANGES_REQUESTED** [review]

File: .travis.yml [hunk 3]
```diff
@@ -27,6 +28,7 @@ before_install:
 
 install:
   - pip install tox
+  - if [ "$TOX_ENV" == "validate-test-tools" ]; then apt-get install libxml2-utils; fi
```
Comment (nsoranzo) [inline .travis.yml]: Now the Travis build fails because apt-get requires sudo privileges.

**Comment by nsoranzo** [comment]
Thanks @jmchilton, that's really nice!

**Pull request merged**